src/Data/Lens/Traversal.purs

```diff
@@ -1,23 +1,47 @@
 -- | This module defines functions for working with traversals.
 
 module Data.Lens.Traversal
-  ( traverse
+  ( traversed
   , traverseOf
+  , sequenceOf
+  , failover
   ) where
 
 import Prelude
 
 import Data.Const
 import Data.Monoid
+import Data.Monoid.Disj
+import Data.Tuple
 import Data.Lens.Types
 import Data.Profunctor.Star
-import Data.Traversable (Traversable)
+import Control.Alternative
+import Control.Plus
+import Data.Traversable (Traversable, traverse)
 import Data.Lens.Internal.Wander (wander)
 
 -- | Create a `Traversal` which traverses the elements of a `Traversable` functor.
-traverse :: forall t a b. (Traversable t) => Traversal (t a) (t b) a b
-traverse = wander
+traversed :: forall t a b. (Traversable t) => Traversal (t a) (t b) a b
+traversed = wander traverse
 
 -- | Turn a pure profunctor `Traversal` into a `lens`-like `Traversal`.
-traverseOf :: forall f s t a b. (Applicative f) => Traversal s t a b -> (a -> f b) -> s -> f t
-traverseOf t f = runStar (t (Star f))
+traverseOf
+  :: forall f s t a b. (Applicative f)
+  => Optic (Star f) s t a b -> (a -> f b) -> s -> f t
+traverseOf t = runStar <<< t <<< Star
+
+-- | Sequence the foci of a `Traversal`, pulling out an `Applicative` effect.
+-- | If you do not need the result, see `sequenceOf_` for `Fold`s.
+sequenceOf
+  :: forall f s t a. (Applicative f)
+  => Optic (Star f) s t (f a) a -> s -> f t
+sequenceOf t = traverseOf t id
+
+-- | Tries to map over a `Traversal`; returns `empty`, if the traversal did
+-- | not have any new focus.
+failover
+  :: forall f s t a b. (Alternative f)
+  => Optic (Star (Tuple (Disj Boolean))) s t a b -> (a -> b) -> s -> f t
```

Comment:
Would it make sense to just return `Maybe`, since `Maybe` can be interpreted this way in any `Alternative`, and it's arguably clearer?